### ISSUE TITLE
Properties of ES classes not appearing in the output

### ISSUE THREAD

**Issue by CedX** [julie-ng/tidy-jsdoc#5]
(sorry to bother you again...)

Using the default template, documenting properties of an ES2015 class is just a matter of adding a `@type` annotation to each property (whether it's an instance property, an instance getter, or a static getter). Properties appear under the "Members" section.

For the `FinderError` class in this file:
https://github.com/cedx/which.js/blob/master/lib/finder.js#L137

The default template output is:
![Default output](https://user-images.githubusercontent.com/1643950/58765666-85903880-8575-11e9-8cfc-2431e39a7bac.jpg)

But when using the Tidy template, the "members" section is skipped:
![Tidy output](https://user-images.githubusercontent.com/1643950/58765678-a8bae800-8575-11e9-9c53-0efde66ec006.jpg)

Whatever class property I'm documenting, they never appear in the generated output.
Another example is the static getter `Finder.isWindows` from the same source file:
https://github.com/cedx/which.js/blob/master/lib/finder.js#L15

It appears in the "Members" section with the default template, but not with the Tidy template.

**Comment by julie-ng** [comment]
## Missing Properties

yes, they were indeed missing. I've added that now and tested it on my library here https://github.com/julie-ng/newtonjs-graph/commit/246ae06baa9a58d587b7c0062bb417d921815b56#diff-893b4b8e5a023704cdb20581e4cf528f

## jsdoc syntax

I think you're having additional issues because of how you are using jsdoc.

Regarding `@type`

> Using the default template, documenting properties of an ES2015 class is just a matter of adding a @type annotation to each property (whether it's an instance property, an instance getter, or a static getter).

Actually, I think you should be using `@property` per the [jsdoc syntax documentation](https://jsdoc.app/tags-property.html) Here is an example from the [jsdoc documentation](https://jsdoc.app/tags-type.html) `/** @type {PropertiesHash} */`

So when you have:

```javascript
    /**
     * The looked up command.
     * @type {string}
     */
    this.command = command;
```

The correct syntax is:

```javascript
    /**
     * The looked up command.
     * @property command {string}
     */
    this.command = command;
```

### Multiple Classes per File??

If you look closely at your screenshot, your page for `Finder` also includes a class `FinderError`, which seems confusing to me. 

<img width="125" alt="Screen Shot 2019-06-20 at 11 08 41" src="https://user-images.githubusercontent.com/1269507/59836437-c943ba00-934b-11e9-80ae-7fb8d5efd884.png">

Are the two related? When debugging your code, I separated them into 2 files to remove that.

**Comment by julie-ng** [comment]
Hmmm, wait... I just came across your usage of `@type` here https://github.com/jsdoc/jsdoc/issues/973

I still need to figure out why the static getters don't appear. But that syntax boggles my mind....

**Comment by CedX** [comment]
> If you look closely at your screenshot, your page for Finder also includes a class FinderError, which seems confusing to me.

The two classes are defined in the same JS file, maybe that's why they are in the same resulting HTML file.

The JSDoc documentation is not very clear. `@property` is used to document an object property (but I'm documenting a class instance property, not an object one). `@type` is used to document the type of value that a symbol may contain: it could be used with virtually anything (any variable, property, expression or function parameter).

From this page (https://jsdoc.app/tags-type.html) : "If you have documented the symbol, JSDoc creates a link to the documentation for that symbol." It's still not clear what a symbol is in terms of JSDoc.

From this page (https://jsdoc.app/tags-property.html) : "The `@property` tag is a way to easily document a list of static properties of a class, namespace or other object." ...but I'm documenting an instance property, not a static one.

The JSDoc documentation needs clarification :smile:

**Comment by chriseaton** [comment]
Just to chime in, I also encountered this issue unexpectedly. Using the default jsdoc output results in the @type members of an ES class listing, but once I use this template, they no longer output. I agree the jsdoc ... documentation... is probably a little wobbly in this case, but I'd agree w/ cedx's approach as that's what I've been doing for a while as well to define properties on classes as well.

----

Edit: I think perhaps the issue that for many of us, just outputting the members of class are tbe, rather than explicity properties which is what the tidy template currently does. One way maybe to address this would be to output all members that are _not_ tagged as properties just under a Members section.

**Comment by julie-ng** [comment]
Found and fixed it. It will look like this in the upcoming release:

![Screen Shot 2020-03-23 at 07 43 26](https://user-images.githubusercontent.com/1269507/77289312-06254e80-6cda-11ea-9ed4-00820398e756.png)

**Comment by julie-ng** [comment]
Fixed now in [v1.3.0 release](https://www.npmjs.com/package/tidy-jsdoc). If the problem persists, feel free to re-open this issue.
